src/ui/RealmBrowser/parsers.ts

```diff
@@ -36,6 +36,22 @@ export const parseObjectId = (
   }
 };
 
+export const parseDecimal128 = (
+  value: string,
+  property: Realm.ObjectSchemaProperty,
+) => {
+  if (value === '' && property.optional) {
+    return null;
+  } else {
+    try {
+      // Note: thousand separators are not supported by Decimal128, so we can help out the user by converting ',' to '.'.
+      return Decimal128.fromString(value.replace(',', '.'));
+    } catch (_) {
+      throw new Error(`"${value}" is not a proper ${property.type}`);
```

Comment:
Does the error message contains a valuable hint which might be useful for the user?

Reply:
Nope, pretty close to our own generic: `X not a valid Decimal128 string`.